docs/docs/concepts/Sync_Async.md

```diff
@@ -0,0 +1,204 @@
+---
+title: "Sync & Async"
+linkTitle: "Sync & Async"
+weight: 3
+description: >
+    Sync and Async Functions
+---
+
+By default, each time a function is invoked, it is invoke synchronously. This means that the atomicity property is promised (no other commands will be invoke or Redis while RedisGears function is running). Atomicity property has some great advantages:
+
+* You can update multiple keys at once and be sure any other client will see the entire update (and not partial updates).
+* You can be sure the data in Redis are not changed while processing it.
+
+On major disadvantage of the atomicity property is that during the entire invocation Redis is blocked and can not serve any other clients.
+
+Triggers and Functions attempt to give a better flexibility to the function writer and allow to invoke function on the background. When function is invoke on the background it can not touch the Redis key space, To touch the Redis key space from the background, the function must block Redis and enter an atomic section where the atomicity property is once again guaranteed.
+
+Triggers and Functions function can go to the background by implement the function as a JS Coroutine and use `registerAsyncFunction`. The Coroutine is invoked on a background thread and do not block the Redis processes. Example:
+
+```js
+#!js api_version=1.0 name=lib
+
+redis.registerAsyncFunction('test', async function(){
+    return 'test';
+});
+```
+
+The above function will simply return `test`, but will run on a background thread and will not block Redis (when running the function, Redis will be able to accept more commands from other clients).
```

Comment:
The simple function shown above returns the string `test` and runs on a background thread that will not block Redis. This allows Redis to continue accepting commands from other clients while the function is running.